### ISSUE TITLE
Fit-to-markers update problems

### ISSUE THREAD

**Issue by emersonbottero** [GoogleWebComponents/google-map#210]
01 - When I dynamic change the quantity of markers to 1 the fit works but the zoom set in the element stop working.
02 - And when the quantity is 1 and you change the location of the marker the auto-center doesn't work.

In my app I have 3 paper-tabs , 2 of them have only one marker but in different locations, switching between then reproduces the second case.

**Comment by emersonbottero** [comment]
Hi @ebidel, Can you confirm if it is a bug?

http://kopempack.com.br/app/
01 - don't select any city/Cidade
02 - change to map view (map icon o toolbar)
03 - switch between tabs 1 and 3.

P.S. Switching between tab 2 and any other tabs works fine.

**Comment by ebidel** [comment]
Looks like it, but it would be more helpful if you could share a reduced test case that repos the issue.

**Comment by emersonbottero** [comment]
@edibel  I tried to make as simple as possible but i don't know other way to dynamically change the number of marks so I used iron-ajax. I'm still learning, started 3 months ago! :D

To reproduce just switch between tabs 1 and 3.

http://kopempack.com.br/test/

Just notice that when you click in the marker nether the image or the title is updated but maybe i need to force it somewhere.

P.S.: if you refresh and don't select any tab the images and title are ok.

**Comment by ebidel** [comment]
It looks like the marker is just outside the map viewport when you click between 1/3. Is that the case to repo this?

**Comment by emersonbottero** [comment]
Yes, shouldn't the fit-to-markers center the marker? I'm doing an App where
people can find events,  the user may think there is none if is not in the
view.
And there is the problem with the markers showing the wrong data. I think
is a google-map bug because when I just list the title outside of it it
comes OK.

2015-10-06 22:52 GMT-03:00 Eric Bidelman <notifications@github.com>:

**Comment by ebidel** [comment]
Yes it should. Something it not being observed correctly. I was just asking if the case where this happens is only for makers just outside the map viewport? Or does it also happen when the marker is far away?

**Comment by emersonbottero** [comment]
It happens in both cases. The difference is that if it is in the viewport
the user will not thing that there are no results. The fit to marker
doesn't work but he can see the marker anyway.


2015-10-07 15:32 GMT-03:00 Eric Bidelman <notifications@github.com>:

**Comment by emersonbottero** [comment]
Hi again @ebidel 

I tried to replace dom -repeat with iron-list and the error continues. I'm not sure if this is a problem with google-map because when I console.log the map element it shows the correct values. In the Image the correct Value is "PASSAPORT SURF ECO FESTIVAL" as you can see in the console.
But in the html code the marker with index 0 keep the data no matter if the list changes or not. The first list of markers had "Marido Matriz e Filial" as index 0. The Title updates correct, only the content is wrong.

If you still can't figure it out I'll try to make the map only with js. Let me know so i can continue the app.

Thanks.

![image](https://cloud.githubusercontent.com/assets/12377432/10595635/2fef08a8-76b9-11e5-9f16-2a4ff67245ef.png)

**Comment by ebidel** [comment]
@emersonbottero this seems to be something else from your original bug report. Possibly related to the switch to iron-list. It's getting very hard to follow what we should be looking at when the demo/test apps are keep changing and the bug descriptions are seemingly unrelated.

If there are several bugs, please post **reduced test case** for each. Preferably on jsbin using polygit.org and not an entire app :)

**Comment by emersonbottero** [comment]
is the exactly same problem as #119 . so I'm closing it as duplicated.
